# Hackbio-cancer-internship-stage4
# NGS Workflow Project

## Authors (@slack)
- **Abdullah Ibrahim Ali** (@Abdullah108)
- **Kareem Elsayed Saad AbdelGawad** (@Kareem_101)

## Project Overview
This project involves the implementation of a Next-Generation Sequencing (NGS) analysis workflow using various bioinformatics tools. Our goal is to process multiple datasets efficiently and perform quality control, trimming, alignment, and variant calling.

## Resources
- **FastQC**: A tool for quality control of high-throughput sequencing data.
- **FastP**: A fast all-in-one preprocessing tool for fastq files.
- **BWA**: A software package for mapping low-quality sequence reads against a large reference genome.
- **Samtools**: A suite of programs for interacting with high-throughput sequencing data.
- **Bcftools**: Tools for variant calling and manipulating VCF and BCF files.

## Used Datasets
- **Reference Genome**: [Reference Genome](https://raw.githubusercontent.com/josoga2/yt-dataset/main/dataset/raw_reads/reference.fasta)
- **ACBarrie**: 
  - [Forward Reads](https://github.com/josoga2/yt-dataset/raw/main/dataset/raw_reads/ACBarrie_R1.fastq.gz)
  - [Reverse Reads](https://github.com/josoga2/yt-dataset/raw/main/dataset/raw_reads/ACBarrie_R2.fastq.gz)
- **Alsen**: 
  - [Forward Reads](https://github.com/josoga2/yt-dataset/raw/main/dataset/raw_reads/Alsen_R1.fastq.gz)
  - [Reverse Reads](https://github.com/josoga2/yt-dataset/raw/main/dataset/raw_reads/Alsen_R2.fastq.gz)
- **Baxter**: 
  - [Forward Reads](https://github.com/josoga2/yt-dataset/raw/main/dataset/raw_reads/Baxter_R1.fastq.gz)
  - [Reverse Reads](https://github.com/josoga2/yt-dataset/raw/main/dataset/raw_reads/Baxter_R2.fastq.gz)
- **Chara**: 
  - [Forward Reads](https://github.com/josoga2/yt-dataset/raw/main/dataset/raw_reads/Chara_R1.fastq.gz)
  - [Reverse Reads](https://github.com/josoga2/yt-dataset/raw/main/dataset/raw_reads/Chara_R2.fastq.gz)
- **Drysdale**: 
  - [Forward Reads](https://github.com/josoga2/yt-dataset/raw/main/dat
## Aims
- To implement an NGS workflow for processing and analyzing sequencing data.
- To ensure high-quality data analysis through quality control and preprocessing steps.
- To perform variant calling to identify genetic variants in the samples.

## Used Programming Languages
- **Bash**: For scripting and automating the NGS workflow.
